src/zope/interface/registry.py

```diff
@@ -38,6 +41,129 @@ def notify(*arg, **kw): pass
 from zope.interface._compat import STRING_TYPES
 
 
+class _UnhashableComponentCounter(object):
```

Comment:
The next line has a comment for that purpose, in lieu of a docstring (since it's not a public class).